aws/usageReports/ebs/monthlyReport.go

```diff
@@ -0,0 +1,263 @@
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package ebs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/trackit/jsonlog"
+	"gopkg.in/olivere/elastic.v5"
+
+	taws "github.com/trackit/trackit-server/aws"
+	"github.com/trackit/trackit-server/aws/usageReports"
+	"github.com/trackit/trackit-server/config"
+	"github.com/trackit/trackit-server/errors"
+	"github.com/trackit/trackit-server/es"
+)
+
+// getElasticSearchEbsSnapshot prepares and run the request to retrieve the a report of an snapshot
+// It will return the data and an error.
+func getElasticSearchEbsSnapshot(ctx context.Context, account, snapshot string, client *elastic.Client, index string) (*elastic.SearchResult, error) {
+	l := jsonlog.LoggerFromContextOrDefault(ctx)
+	query := elastic.NewBoolQuery()
+	query = query.Filter(elastic.NewTermQuery("account", account))
+	query = query.Filter(elastic.NewTermQuery("snapshot.id", snapshot))
+	search := client.Search().Index(index).Size(1).Query(query)
+	res, err := search.Do(ctx)
+	if err != nil {
+		if elastic.IsNotFound(err) {
+			l.Warning("Query execution failed, ES index does not exists", map[string]interface{}{
+				"index": index,
+				"error": err.Error(),
+			})
+			return nil, errors.GetErrorMessage(ctx, err)
+		} else if cast, ok := err.(*elastic.Error); ok && cast.Details.Type == "search_phase_execution_exception" {
+			l.Error("Error while getting data from ES", map[string]interface{}{
+				"type":  fmt.Sprintf("%T", err),
+				"error": err,
+			})
+		} else {
+			l.Error("Query execution failed", map[string]interface{}{"error": err.Error()})
+		}
+		return nil, errors.GetErrorMessage(ctx, err)
+	}
+	return res, nil
+}
+
+// getSnapshotInfoFromEs gets information about an snapshot from previous report to put it in the new report
+func getSnapshotInfoFromES(ctx context.Context, snapshot utils.CostPerResource, account string, userId int) Snapshot {
+	var docType SnapshotReport
+	var snap = Snapshot{
+		SnapshotBase: SnapshotBase{
+			Id:          snapshot.Resource,
+			Description: "N/A",
+			State:       "N/A",
+			Encrypted:   false,
+			StartTime:   time.Time{},
+		},
+		Tags:  make([]utils.Tag, 0),
+		Cost: 0,
+		Volume: Volume{
+			Id:    "N/A",
+			Size:  -1,
```

Comment:
Make sure to run `gofmt` to have the right alignment for data in these structures

Reply:
Fixed